### PR TITLE
Handle deprecated Matroska Elements

### LINKED ISSUE #117: Consider removing outdated/deprecated Matroska Elements
See the latest [Matroska draft](https://www.ietf.org/archive/id/draft-ietf-cellar-matroska-16.html#name-annex-a-historic-deprecated):

> Many of the elements that were then defined are not found in any known files but were part of public specs. DivX also had a few custom elements that were designed for custom features.

There are quite a few Elements listed there that were still in the spec when they were added to this code base (the comments refer to draft v03), but are now removed. Some examples are BlockVirtual, ReferenceVirtual, SilentTracks and all the DivX TrickTrack extension elements.

Maybe there should be a discussion on how to handle Matroska/EBML versioning in general. I can see some ways to handle Elements being removed from the spec.

1. Just keep all historic, deprecated elements in the code base. This will lead to a bit of bloat, but will lead to the best compatibility. It would also mean a potentially bigger maintenance effort.
2. Simply remove the Elements from the code as soon as a new draft/spec version drops them. This means potentially breaking compatibility with old files that rely on these features existing. However, if we assume that the Elements being dropped from the spec will only be Elements that are almost never used, this could turn out to be a non-problem in reality.
3. Consider dropping Elements on a case-by-case basis. This would probably mean going down option 2, but keeping Elements that we consider to be "important enough" to not simply delete.

One other thing I'm not 100% sure about is this: What happens if the parser encounters an unknown Element ID while parsing? If we go down route 2 or 3, and the parser *does* encounter an Element that it doesn't know, will it just completely fail parsing that Element? I'd argue that (*at least* for a list of known, deprecated IDs), the parser should just skip the deprecated Element instead of rejecting the entire file. I think that it currently errors out, but I am not sure. Something to test (and maybe introduce tests for).

(I'll soon add a list of Elements that are currently not in the spec, but still in the code base. IMO we should decide on these Elements too, to get a "clean slate". Discussion can start before that though)

### PR DESCRIPTION
**Pull request by FreezyLemon**
Closes #117.

There was a bug in `skip_master` (now `skip_element`) where it didn't skip all of the Element data which was easy to fix.

Also removed some serialization functions that aren't used anymore.

### CODE REVIEW

File: src/permutation.rs [hunk 1]
```diff
@@ -116,9 +116,14 @@ macro_rules! permutation_trait_impl(
           // Have all parsers (including void) failed?
           if l == input.len() {
             // Skip unknown Element if possible.
-            if let Error { id: _, kind: crate::ebml::ErrorKind::MissingElement } = err {
-              if let Ok((i, id)) = crate::ebml::skip_master(input) {
-                log::warn!("Skipped unknown Element 0x{id:X}");
+            if let Error { id: _, kind: $crate::ebml::ErrorKind::MissingElement } = err {
+              if let Ok((i, id)) = $crate::ebml::skip_element(input) {
```
Comment (Luni-4) [inline src/permutation.rs:120]: Is it possible to use `if let` chains here?

They should be stabilized starting from Rust 1.64 https://github.com/rust-lang/rust/pull/94927

**Review by Luni-4 — APPROVED** [review]
Thanks a lot! This is really fantastic! 

Just a simple nit

File: src/permutation.rs (hunk 1 above)
Comment (FreezyLemon) [reply src/permutation.rs:120]: Oops, doesn't work yet. Some stuff is still not stable:
https://github.com/rust-lang/rust/issues/53667

```
 1  error[E0658]: `let` expressions in this position are unstable                                                          ▐
    --> src/permutation.rs:119:16                                                                                          ▐
     |                                                                                                                     ▐
 119 |               if let Error { id: _, kind: $crate::ebml::ErrorKind::MissingElement } = err                           ▐
     |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                           
```

Comment (Luni-4) [reply src/permutation.rs:120]: Oh, what a pity! Ok, no problem then

**Pull request merged**